fix(docs-infra): make header full-width and panels full-width on mobile screens

ajitsinghkaler opened the pull request:
Fixes #34163

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
In the current behavior, the width of the header is lesson events page in mobile screen and I made resources full screen on the mobile screen

Issue Number: #34163

## What is the new behavior?
The header is full screen and resources are full screens on mobile.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

ajitsinghkaler commented:
@gkalpak can you please help me and tell me why the integration test is failing.

gkalpak commented on aio/src/styles/1-layouts/_marketing-layout.scss:
```diff
@@ -411,6 +411,7 @@ div[layout=row]{
   background-color: lighten($blue, 10);
   margin-top: 64px;
   padding: 32px;
+  width:100%;
```
Please, put a space after `:`.

gkalpak replied:
Also, can you explain what difference does this make? `.marketing-banner` is only set on `<header>` (afaict), which is a block element by default. So, `width`, shouldn't make a difference. Did I miss something?

ajitsinghkaler replied:
In the current scenario if the table in the body of the docs takes more than screen width the header does not take 100% of the screen. That is if the screen is horizontally scrollable on mobile screens then the header does not take full width making width:100% lets it take full width in those scenarios also. I've attached a screenshot for reference.

![Angular - EVENTS (1)](https://user-images.githubusercontent.com/39260684/70069761-4fad3900-1618-11ea-8c87-e1301cb4f52f.png)

gkalpak replied:
I see. I don't think this is the right fix (because I still see it not taking up 100% of the scroll width.
I believe you need to set `overflow: auto` (or `overflow-x: auto`) to the [`<article>` element][1] there.

[1]: https://github.com/angular/angular/blob/3eedf884fa50124ff0f5dfe9be4aa2b6150dc957/aio/content/marketing/events.html#L5

ajitsinghkaler replied:
but I've checked it in the build it is working correct I've attached screenshots and link of the build page for your reference. Link: https://pr34188-09ee065.ngbuilds.io/events Please use this link in the mobile or responsive web version.

![Angular - EVENTS](https://user-images.githubusercontent.com/39260684/70114442-7c963600-1682-11ea-88f3-0974e4142d47.png)

ajitsinghkaler replied:
![Desktop screenshot (2)](https://user-images.githubusercontent.com/39260684/70114586-f9291480-1682-11ea-8136-0aa514ff3607.png)

gkalpak replied:
Your change does fix the issue on some screen sizes, but not all. If the screen is smaller (e.g. in DevTools choose iPhone 5) there is still an empty space on the right of the header:

![image](https://user-images.githubusercontent.com/8604205/70130614-82434a00-1689-11ea-94b5-90fbdee6e460.png)

Do you have any objection against my suggestion in https://github.com/angular/angular/pull/34188#discussion_r353307115?

ajitsinghkaler replied:
No I didn't have any objection it was just width:100% was working in my pc.

gkalpak commented on aio/src/styles/2-modules/_resources.scss:
```diff
@@ -1,5 +1,8 @@
 .showcase {
   width: 80%;
+  @media (max-width: 600px) {
+    width:100%;
```
Please, put a space after `:`.

ajitsinghkaler replied:
I've put a space.

Here right now we have put resources 80% on desktop leaving room for a sidebar if needed to be added later. but in mobile screens, the sidebar will not be added anyways. So I used the whole space and made it go 100%  on small screens. Attaching a screenshot for reference.

![Angular - EXPLORE ANGULAR RESOURCES (1)](https://user-images.githubusercontent.com/39260684/70070039-e2e66e80-1618-11ea-98ae-ac9d98b9a851.png)

mary-poppins commented:
You can preview 416c06e at https://pr34188-416c06e.ngbuilds.io/.
You can preview 09ee065 at https://pr34188-09ee065.ngbuilds.io/.

gkalpak reviewed (changes requested):
Thx, @ajitsinghkaler! Can you, please, update the commit message to `fix(docs-infra): ...` and also add some more info on what exactly it is fixing (e.g. the `events` and `resources` pages)?

mary-poppins commented:
You can preview 3eedf88 at https://pr34188-3eedf88.ngbuilds.io/.

ajitsinghkaler commented:
@gkalpak now you can have a look I attached screenshots for what changes does this PR make.

gkalpak commented:
Thx, @ajitsinghkaler. Can you also take care of the second part of https://github.com/angular/angular/pull/34188#pullrequestreview-326062811?

ajitsinghkaler commented:
@gkalpak I've integrated the changes you suggested and also changes the commit message please have a look

mary-poppins commented:
You can preview b934e2c at https://pr34188-b934e2c.ngbuilds.io/.

gkalpak commented on aio/src/styles/1-layouts/_marketing-layout.scss:
```diff
@@ -439,3 +439,7 @@ div[layout=row]{
 }
+
+.events-container{
+  overflow-x:auto;
```
```suggestion
  overflow-x: auto;
```

ajitsinghkaler committed:
fix(docs-infra): make header full-width and panels full-width on mobile screens

On events page the header was not able to take full width when body exceeds viewport width of the screen So made the below body go overflow-x auto and resources page was taking 80% of the width which is okay on desktop but on mobile it should take 100% width put a media quer for it.

Fixes #34163

mary-poppins commented:
You can preview 9f2ae82 at https://pr34188-9f2ae82.ngbuilds.io/.

ajitsinghkaler commented:
@gkalpak I've made the changes please approve the pull request.

gkalpak commented:
**merge-assistance:** This is a docs-infra only change (it doesn't touch marketing content). My approval should be enough.
